compile/compile.go

```diff
@@ -210,17 +219,72 @@ func (c *Compiler) WithMetadata(metadata *map[string]interface{}) *Compiler {
 	return c
 }
 
+func addEntrypointsFromAnnotations(c *Compiler, ar []*ast.AnnotationsRef) error {
+	for _, ref := range ar {
+		scope := ref.Annotations.Scope
+		entrypoint := ast.EmptyRef()
+		if ref.Annotations.Entrypoint && (scope == "rule" || scope == "package") {
+			// Build up the entrypoint path from component parts.
+			if p := ref.GetPackage(); p != nil {
+				entrypoint = entrypoint.Concat(p.Path)
+			}
+			if r := ref.GetRule(); r != nil {
+				entrypoint = entrypoint.Append(ast.StringTerm(r.Head.Name.String()))
+			}
+			// Get a slash-based path, as with a CLI-provided entrypoint.
+			slashPath, err := entrypoint.Ptr()
+			if err != nil {
+				return err
+			}
+			// Add new entrypoints to the appropriate places.
+			c.entrypoints = c.entrypoints.Append(slashPath)
+			c.entrypointrefs = append(c.entrypointrefs, ast.NewTerm(entrypoint))
+		}
+	}
+
+	return nil
+}
+
 // Build compiles and links the input files and outputs a bundle to the writer.
 func (c *Compiler) Build(ctx context.Context) error {
 
 	if err := c.init(); err != nil {
 		return err
 	}
 
+	// Fail early if not using Rego annotation entrypoints.
+	if err := c.checkNumEntrypoints(false); err != nil {
```

Comment:
I'm a little confused by the interplay between the `bool` argument here, and c.useRegoAnnotationEntrypoints` -- I suspect that it could be rearrange a bit to be more straightforward, but I trust that you've tried that already 🙃